labcore/instruments/qick/qick_sweep.py

```diff
@@ -65,34 +65,58 @@ def collect(self, *args, **kwargs):
         """
         Get the measurement data.
         Note that one can overload the given acquire function if one needs
-        to perform other specific tasks. e.g. Needs to plot each non-averaged 
+        to perform other specific tasks. e.g. Needs to plot each non-averaged
         points in the I-Q plane for a readout fidelity experiment.
         Assumptions
         * Given DataSpecs are either independent, dependent, or ComplexQICKData.
         """
+        # TODO: How can I extend this to multiple measurement rounds? (e.g. active reset)
 
         # Run the program
         data = self.communicator["qick_program"].acquire(self.config.soc)
-        
-        # Parse data according to annotated DataSpecs
-        return_data = ()  # tuple to be yielded
-
-        if len(self.specs) != len(data):
-            NotImplementedError("DataSpecs and returned data have different lengths.")
 
-        for i, ds, da in enumerate(zip(self.specs, data)):
-            if isinstance(ds, independent) or isinstance(ds, dependent):
-                return_data.append(da)
-            elif isinstance(ds, ComplexQICKData):
-                i_data = da[0]
-                q_data = da[1]
-
-                if (i_data is not None) and (q_data is not None):
-                    return_data.append(i_data + 1j*q_data) 
+        # Parse data according to annotated DataSpecs
+        return_data = {}  # tuple to be yielded
+        if len(data) == 2:  # No QICK-internal dependent variables
+            ds = self.specs[0]
+            return_data[ds.name] = data[0][0] + 1j*data[1][0]
+        else:
+            measIdx = 0  # To specify the index of the measured data to return
+            sweepIdx = 0  # To specify the index of the sweep variable to return
+            isNot1D = isinstance(self.communicator["qick_program"].get_expt_pts(), list)
+            for ds in self.specs:
+                if isinstance(ds, ComplexQICKData):
+                    return_data[ds.name] = data[1][measIdx][0] + 1j*data[2][measIdx][0]
+                    measIdx += 1
+                elif isNot1D:
+                    return_data[ds.name] = self.communicator["qick_program"].get_expt_pts()[sweepIdx]
+                    sweepIdx += 1
                 else:
-                    NotImplementedError("ComplexQICKData requires both I and Q data.")
+                    return_data[ds.name] = self.communicator["qick_program"].get_expt_pts()
+                    sweepIdx += 1
+
+            # Reformat the independent variables
```

Comment:
Maybe add another line here as to how and why you are reformatting variables? Might be worth to just write a little paragraph in the docstring explaining that.

Reply:
Keep it as comments here since the user doesn't really need to see that to use the code

Reply:
I am reformatting the variables so that xArray and plottor can correctly recognize which axis is the axis being swept. Without reformatting the swept variable, the program won't be able to recognize the axis being swept automatically.

Reply:
yes, can you add exactly that in the actual code as a comment under line 98. I don't think it would be obvious to someone else reading the code from the get go